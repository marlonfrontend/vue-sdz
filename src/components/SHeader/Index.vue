<template>
  <s-collapsible class="s-header" no-header v-bind="$attrs" v-on="$listeners">
    <slot />
  </s-collapsible>
</template>

<script>
import SCollapsible from '../../../src/components/SCollapsible/Index.vue'

export default {
  name: 'SHeader',

  components: { SCollapsible }
}
</script>

<style lang="scss">
@import "./src/styles/_index.scss";

.s-header {
  top: 0;
  position: sticky;
  z-index: $z-index-1;
}
</style>
