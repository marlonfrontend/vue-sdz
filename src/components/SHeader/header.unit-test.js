import { mount } from '@vue/test-utils'

import SHeader from './Index.vue'

/* eslint-disable */
describe('SHeader', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(SHeader, {
      propsData:{
        isOpened: true
      },
      slots:{
        default: '<div>Item da Header</div>'
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check collapsible class', () => {

    const text = cmp.find('.s-collapsible')
    expect(text.exists()).toBe(true)

  })

  test('check body text', () => {

    const text = cmp.find('div > div > div')
    
    expect(text.exists()).toBe(true)
    expect(text.text()).toContain('Item da Header')

  })

})
