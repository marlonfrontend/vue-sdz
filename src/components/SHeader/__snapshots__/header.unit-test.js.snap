// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHeader has the expected html structure 1`] = `
<div
  class="s-collapsible s-header"
  isopened="true"
>
  <div
    class="header"
  >
    <!---->
  </div>
   
  <div
    class="wrapper"
    style="opacity: 1; height: 0px;"
  >
    <div>
      Item da Header
    </div>
  </div>
</div>
`;
